tests/pkg_tiny-asn1/main.c

```diff
@@ -0,0 +1,207 @@
+
+    if (der_decode(cms_data, sizeof(cms_data), &cms, asn1_objects, asn1_object_count) < 0) {
+        printf("ERROR: Could not parse the data.\n");
+        return 1;
+    }
+
+    /* Dump the data */
+    print_asn1(&cms, 0);
+
+    /* Since we know this is CMS data, we can try to interpret it. */
+    if (cms.type != 0x30) {
+        printf("ERROR: The outer type is not a SEQUENCE.\n");
+        return 1;
+    }
+
+    asn1_tree *content_type = cms.child;
+    if (content_type == NULL || content_type->type != 0x06) {
+        printf("ERROR: No ContentType information available.\n");
+        return 1;
+    }
+    printf("ContentType: ");
+    if (memcmp(content_type->data, (uint8_t[]){ 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07, 0x06, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07, 0x06 }, content_type->length) != 0) {
+        printf("ERROR: ContentType is not of type encryptedData.\n");
+        return 1;
+    }
+    printf("encryptedData\n");
+
+    asn1_tree *encrypted_data = content_type->next->child;
+    if (encrypted_data == NULL || encrypted_data->type != 0x30) {
+        printf("ERROR: EncryptedData not availavble.\n");
+        return 1;
+    }
+
+    asn1_tree *cms_version = encrypted_data->child;
+    if (cms_version == NULL || cms_version->type != 0x02 || cms_version->length != 1) {
+        printf("ERROR: CMSVersion not availavble.\n");
+        return 1;
+    }
+    uint8_t version = cms_version->data[0];
+    printf("CMSVersion: %d\n", version);
+
+    asn1_tree *encrypted_content_info = cms_version->next;
+    if (encrypted_content_info == NULL || encrypted_content_info->type != 0x30) {
+        printf("ERROR: EncryptedContentInfo not availavble.\n");
+        return 1;
+    }
+
+    asn1_tree *encrypted_content_info_content_type = encrypted_content_info->child;
+    if (encrypted_content_info_content_type == NULL || encrypted_content_info_content_type->type != 0x06) {
+        printf("ERROR: ContentType of EncryptedContentInfo not availavble.\n");
+        return 1;
+    }
+    if (memcmp(encrypted_content_info_content_type->data,
+               (uint8_t[]){ 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07, 0x01 }, encrypted_content_info_content_type->length) != 0) {
+        printf("ERROR: ContentType of EncrytptedContentInfo is not of type pkcs#7.\n");
+        return 1;
+    }
+    printf("ContentType of EncryptedContentInfo: pkcs#7\n");
+
+    asn1_tree *content_encryption_algorithm = encrypted_content_info_content_type->next;
+    if (content_encryption_algorithm == NULL) {
+        printf("ERROR: ContentEncryptionAlgorithm not available.\n");
+        return 1;
+    }
+
+    if (content_encryption_algorithm->type == 0x30) {
+        /* extended definition of ContentEncryptionAlgorithmIdentifier by RFC5911 */
+        asn1_tree *encryption_algorithm_identifier = content_encryption_algorithm->child;
+        if (encryption_algorithm_identifier == NULL || encryption_algorithm_identifier->type != 0x06) {
```

Comment:
(at least this is the only point where the static code analyzer is complaining, but I honestly don't understand why only here :/